### DISCUSSION TITLE
Request: Stirling-PDF

### DISCUSSION THREAD

**Discussion by Brixxor** [tteck/Proxmox#1887] [Ideas]
Hi this could be an interesting addition to your selfhosted portfolio:

https://github.com/Frooodle/Stirling-PDF

Thnx for your work

**Comment by ajtatum** [comment]
Would be great to have this!

**Comment by Frooodle** [comment]
Hey I'm the maintainer of Stirling pdf I have no experience with proxmox but let me know if I can help in any way with this!

**Reply by tteck** [reply]
@Frooodle I noticed that "App Version" isn't populating, is this a known issue?

![Screenshot 2024-01-08 5 03 23 AM](https://github.com/tteck/Proxmox/assets/21342670/327e9c7d-f24e-45ff-bc33-34c844d6bbb7)

**Reply by Frooodle** [reply]
Known issue with latest release

**Comment by tteck** [comment]
Ready for testing
```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/stirling-pdf.sh)"
```

**Reply by tteck** [reply]
As soon as I get confirmation that everything is working, I will add it to the website.

**Reply by tteck** [reply]
What's a better test than going live? It's on the site.

**Reply by di5cord20** [reply]
Works for me.  Thanks for another great script!

**Reply by quinton-c** [reply]
Worked great for me too. These scripts are amazing!

**Comment by di5cord20** [comment]
@tteck , how should we update this container?  The usual "update" command ??
![image](https://github.com/tteck/Proxmox/assets/140566995/dcd8c897-0efc-4958-b030-6840f80cfd7d)

**Reply by tteck** [reply]
Yes, use the `update` command. That "Update available" is just a link to Releases, it's NOT an indication that an actual update is available.

**Reply by tteck** [reply]
That should be renamed to "Check releases"

**Reply by di5cord20** [reply]
Thanks.  You typically include details on the update procedure which I didn't see in Stirling-PDF (i.e. `To Update Stirling-PDF, run the command below (or type update) in the LXC Console.`)

**Reply by tteck** [reply]
It's there!

**Reply by Frooodle** [reply]
> That should be renamed to "Check releases"
> 

Actually it does onto show if there is a update it is based on current version and compared to latest version

As current version is bugged in this latest release to 0.0.0 it checks and displays the button 

Hopefully fixed next patch once I figure out cause for version suddenly not showing

**Reply by tteck** [reply]
@di5cord20 Yesterday, I performed extensive code refactoring on the script. It might be prudent to establish a new LXC to ensure that all components are current and functioning correctly.

**Reply by di5cord20** [reply]
@tteck , I deleted my previous container and just installed again.  Previously it took a long time to install all dependencies, etc.  This time it finished almost immediately, doesn't show any errors, but I believe it didn't work properly.  I'll try again but wanted to share here in case.

![image](https://github.com/tteck/Proxmox/assets/140566995/7f370867-c0e3-438e-b600-d6a70151c8cd)

**Reply by tteck** [reply]
Clear browser cache. 
The script's name changed.

**Reply by tteck** [reply]
```
   _____ __  _      ___                 ____  ____  ______
  / ___// /_(_)____/ (_)___  ____ _    / __ \/ __ \/ ____/
  \__ \/ __/ / ___/ / / __ \/ __ `/___/ /_/ / / / / /_
 ___/ / /_/ / /  / / / / / / /_/ /___/ ____/ /_/ / __/
/____/\__/_/_/  /_/_/_/ /_/\__, /   /_/   /_____/_/
                          /____/
Using Advanced Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 1
Using Root Password: Automatic Login
Container ID: 111
Using Hostname: stirling-pdf
Using Disk Size: 8
Allocated Cores: 2
Allocated RAM: 2048
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using MAC Address: AE:1A:60:07:B7:59
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a Stirling-PDF LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using local-lvm for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 111 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.253 
 ✓ Internet Connected
 ✓ DNS Resolved github.com to 140.82.114.3
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Installed LibreOffice Components
 ✓ Installed Python Dependencies
 ✓ Installed Azul Zulu
 ✓ Installed JBIG2
 ✓ Installed Language Packs
 ✓ Installed Stirling-PDF v0.19.0
 ✓ Created Service
 ✓ Customized Container
 ✓ Cleaned
 ✓ Completed Successfully!

Stirling-PDF should be reachable by going to the following URL.
         http://192.168.86.253:8080  
```

**Reply by di5cord20** [reply]
Worked like a charm.  Thanks again.

**Comment by di5cord20** [comment]
> It's there!

My mistake.  I leave your script page open and likely forgot to refresh.  Thanks

**Comment by felersgit** [comment]
I keep getting this error when I attempt run this script.  I have ran it a couple of times and results in the same error.  I also bumped up the cpu's/ram but to no avail.

```
[ERROR] in line 62: exit code 0: while executing command $STD git clone https://github.com/agl/jbig2enc /opt/jbig2enc
++ [[ 62 -eq 24 ]]
+ exit
```

**Reply by tteck** [reply]
No issues here. Blocked by something?
```
root@debian:~# git clone https://github.com/agl/jbig2enc /opt/jbig2enc
Cloning into '/opt/jbig2enc'...
remote: Enumerating objects: 326, done.
remote: Counting objects: 100% (149/149), done.
remote: Compressing objects: 100% (36/36), done.
remote: Total 326 (delta 125), reused 117 (delta 113), pack-reused 177
Receiving objects: 100% (326/326), 1.40 MiB | 5.78 MiB/s, done.
Resolving deltas: 100% (188/188), done.
```

**Reply by felersgit** [reply]
It was DNS issue, weird I never get that with these scripts.  Oh well the script works now, only problem after running several tests with the program itself is that when you try to convert a txt or doc file to a pdf it has issues with the libreoffice install.  I don't have the error it gives, but it is with the python stuff.  I will try to get it later or if you can do a simple test of converting a txt document to a pdf you will see the error.

**Comment by towerhand** [comment]
Can you try again @tteck? When I tried last night it ended up in an endless loop. I didn't have time to try again.

**Reply by tteck** [reply]
> I didn't have time to try again.

And I do? Everything is working as expected. https://github.com/tteck/Proxmox/discussions/1887#discussioncomment-8066162

**Reply by towerhand** [reply]
Would you mind trying with the verbose option on?

It shows an endless loop printing this message when I try to install it.

+ printf '\b%s' -
-+ sleep 0.1
+ true
+ printf '\b%s' '\'
\+ sleep 0.1
+ true
+ printf '\b%s' '|'
|+ sleep 0.1
+ true
+ printf '\b%s' /
/+ sleep 0.1
+ true
+ printf '\b%s' -
-+ sleep 0.1
+ true
+ printf '\

**Reply by tteck** [reply]
That's the spinner in verbose.

**Reply by tteck** [reply]
I'm going to remove `set -x`, this will allow only the `$STD` entries to be shown in verbose mode.

**Reply by tteck** [reply]
Verbose mode has been modified, if you'd like to try again.

**Reply by towerhand** [reply]
Now it works as it should. Thank you!!!

**Reply by tteck** [reply]
Yeah, that cleaned it up a bunch.

**Comment by Chrischi-** [comment]
How about integrating OCR? That doesn't work at the moment.

**Reply by Frooodle** [reply]
There was a update to OCR pathing yesterday, that might be cause

**Reply by Chrischi-** [reply]
That was it. I have created a simple symbolic link as a quick fix.

`ln -s /usr/share/tesseract-ocr/5/tessdata/ /usr/share/tessdata`
